stactask/task.py

```diff
@@ -108,15 +107,29 @@ def __init__(
 
     @property
     def process_definition(self) -> dict[str, Any]:
-        process = self._payload.get("process", {})
+        process = self._payload.get("process", [])
         if isinstance(process, dict):
+            warnings.warn(
+                "`process` is dictionary, use a list instead",
```

Comment:
I think this message is a bit ambiguous, as the dict itself shouldn't change, it just needs to be wrapped in an list. So maybe it better to say something like:

> `process` as a bare dictionary will be unsupported in a future version; wrap it in a list to remove this warning

---

Side note, I think it would be better to use javascript terminology here, as the payload is in json format. So like "map" and "array" instead of "dictionary" and "list". It seems that language is in use in other places though so oh well, it's better to be consistent.

Reply:
Updated the message.

>It seems that language is in use in other places though so oh well, it's better to be consistent.

Agreed on using javascript terminology. Since you also see this as something that should be updated, I'll get a wholesale change into another PR (either a sidecar fix, or standalone PR). For now, I agree that we should stay consistent with existing language ("dictionary" and "list") so we can get this PR merged more quickly to support those waiting on this change.